WIP: DDS-1218-auth_token

dmlond opened the pull request:
basic scaffolding for the ruby application with rspec tests

dmlond committed:
initial commit of dds_md5_reporter application with initial spec

Gemfile adds httparty, rspec, and active_support

dds_md5_reporter
  script with require guard
  initialization
  json_headers
  auth_token
spec/dds_md5_reporter_spec

dmlond committed:
Merge branch 'master' of github.com:Duke-Translational-Bioinformatics/md5_reporter into DDS-1218-auth_token

dmann commented on spec/dds_md5_reporter_spec.rb:
```diff
@@ -0,0 +1,199 @@
+      )
+    }
+
+    describe '.json_headers' do
```
Nit pick: Use `#` for instance methods. So, this should be:
```
describe '#json_headers' do
```

dmlond committed:
based on suggestions during Code Review
- changed positional arguments to required keyword arguments
- spec uses standard '#method' instead of '.method' nomenclature

added usage and exception handling

dmann commented on dds_md5_reporter.rb:
```diff
@@ -4,11 +4,11 @@
-    @agent_key = agent_key
-    @dds_api_url = dds_api_url
+  def initialize(upload_id:, user_key:, agent_key:, dds_api_url:)
+    @upload_id = upload_id || raise(ArgumentError, "missing keywords: upload_id")
```
`ArgumentError` should be raised automatically, I don't think the `|| raise(ArgumentError, ...` part is needed.

dmlond committed:
refactored check for nil arguments to pass back all required arguments
rather than one at a time.

dmlond committed:
tests ArgumentError message

dmann reviewed (approved):
Whatever :shipit:
